examples/usecases/transformers-next-item-prediction.ipynb

```diff
@@ -538,14 +538,6 @@
     "We are now ready to train our model."
```

Comment:
<p>We need to add an MLPBlock after the Input Block for projecting the input output dim to XLNET d_model dim. I recommend to modify the code as below:</p><p> </p><pre class="ql-syntax" spellcheck="false">dmodel=48

mlp_block = mm.MLPBlock(
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; [128,dmodel],
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; activation='relu',
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; no_activation_last_layer=True,
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; )

model = mm.Model(
&nbsp; &nbsp; mm.InputBlockV2(
&nbsp; &nbsp; &nbsp; &nbsp; seq_schema,
&nbsp; &nbsp; &nbsp; &nbsp; embeddings=mm.Embeddings(
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; train_set_processed.schema.select_by_tag(Tags.CATEGORICAL), sequence_combiner=None
&nbsp; &nbsp; &nbsp; &nbsp; ),
&nbsp; &nbsp; ),
&nbsp; &nbsp; mlp_block,
&nbsp; &nbsp; mm.XLNetBlock(d_model=dmodel, n_head=4, n_layer=2,&nbsp;
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; pre=mm.SequentialBlock([mm.SequenceMaskLastInference(), mm.ReplaceMaskedEmbeddings()]),
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; post="inference_hidden_state",
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;),
&nbsp; &nbsp; mm.CategoricalOutput(
&nbsp; &nbsp; &nbsp; &nbsp; train_set_processed.schema.select_by_name(target),
&nbsp; &nbsp; &nbsp; &nbsp; default_loss="categorical_crossentropy",
&nbsp; &nbsp; ),
)
</pre><p>you can change the <code>dmodel</code> as you like. if you update the code like that then you can remove this line from the unit test <code>tb.cells[28].source = tb.cells[28].source.replace("d_model=64", "d_model=40")</code> . The reason it was giving error is bcs MLP block is missing and that creates dimension mismatch.</p><p> Please also note that I updated the pre arg as well.   it is to support inference.   the input sequence and its corresponding mask should be  extended by one position at the end to account for the next-item (target) position.     To do this, you should set SequenceMaskLastInference  as a pre-layer of    <code>ReplaceMaskedEmbeddings()</code> using the sequential-block. we dont have to add that in this example since this is only doing training and eval. you can remove it if you want to.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/models/pull/943/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='27'/>

Reply:
thank you @rnyak, this is great! 🙂 made the changes